packages/widgetbook_core/lib/src/navigation_tree/widgets/navigation_tree_item.dart

```diff
@@ -0,0 +1,165 @@
+import 'package:flutter/material.dart';
+import 'package:widgetbook_core/widgetbook_core.dart';
+
+class NavigationTreeItem extends StatefulWidget {
+  const NavigationTreeItem({
+    super.key,
+    this.level = 0,
+    this.label,
+    this.content,
+    this.iconData,
+    this.icon,
+    this.onTap,
+    this.isExpanded = false,
+    this.onMoreIconPressed,
+  })  : assert(
+          label != null || content != null,
+          'Either a label or content must be provided',
+        ),
+        assert(
+          icon != null || iconData != null,
+          'Either an icon or iconData must be provided',
+        );
+
+  const NavigationTreeItem.category({
+    super.key,
+    required this.label,
+    this.onTap,
+    this.isExpanded = false,
+    this.onMoreIconPressed,
+  })  : level = 0,
+        iconData = Icons.auto_awesome_mosaic,
+        icon = null,
+        content = null;
```

Comment:
are these two lines necessary?

Reply:
Yes since this is the initialization of the final parameters, so it's instead of defining `this.icon`/`this.content` inside the constructor.

Reply:
Ok, no it's fine. It just looks a bit weird 😀